static/js/controllers/reports.js

```diff
@@ -423,13 +423,18 @@ angular.module('inboxControllers').controller('ReportsCtrl',
     };
 
     $scope.setLeftActionBar({
-      exportFn: function() {
+      exportFn: function(e) {
         var exportFilters = _.extendOwn({}, $scope.filters);
         ['forms', 'facilities'].forEach(function(type) {
           if (exportFilters[type]) {
             delete exportFilters[type].options;
           }
         });
+        var a = $(e.target).closest('a')[0];
+        a.classList.add('mm-icon-disabled');
+        $timeout(function() {
+          a.classList.remove('mm-icon-disabled');
+        }, 2000);
```

Comment:
Happy to have changed this, all three solutions are unquestionably fine to me.

For the purposes of my education though: I'm interested in your reasoning here.

I get why angular would be better: in general you don't have to manage the state yourself, angular does it with watchers (though not this time, you are just moving the state management from directly on the DOM to a scope variable); as well as consistency. I figured in this case though we don't really _want_ to codify it as some officially correct thing: however I can see other perspectives on that.

However: I don't get why this linked code is better. It seems identical, but using more jquery? `.classList` has been in Chrome since forever, so it's not a compatibility thing. I'm not sure why what you wrote is preferable to what I wrote.